tests/forms_tests/widget_tests/test_selectdatewidget.py

```diff
@@ -485,3 +485,39 @@ def test_value_omitted_from_data(self):
         self.assertIs(self.widget.value_omitted_from_data({'field_day': '1'}, {}, 'field'), False)
         data = {'field_day': '1', 'field_month': '12', 'field_year': '2000'}
         self.assertIs(self.widget.value_omitted_from_data(data, {}, 'field'), False)
+
+    def test_regexp_date_re(self):
+        # check valid formats
+        # 1 digit for (day & month)
+        self.assertDictEqual(self.widget.format_value('2000-1-1'), {'day': 1, 'month': 1, 'year': 2000})
```

Comment:
`assertEqual` should be fine. See note in the [Python docs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addTypeEqualityFunc), "Note that it’s usually not necessary to invoke these methods directly."